common/src/test/java/io/netty/util/internal/MaxDirectMemorySettingTest.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util.internal;
+
+import org.junit.Ignore;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public final class MaxDirectMemorySettingTest {
+
+  /**
+   * This test tests only one case when the classloader loads the {@link PlatformDependent} class.
+   * This test must be the very first method called otherwise loading the class previous to this, will
+   * initialize it and dismiss this test case.
+   * <p/>
+   * A JUnit test runner that could use different classloader per test method would allow to test cases
+   * where the MaxDirectMemorySetting check is enabled, setting vs not setting the max direct memory.
+   */
+  @Test
+  @Ignore("see test comment")
```

Comment:
maybe we coudl jsut remove it, no need to add a "ignored" test imho

Reply:
Ok will do, I didn't want to remove an existing test ;o)

Maybe to test that this works as expected, we can create an integration test.

Reply:
@rendel For the integ test, I can add the allocator limit (which is the `max_direct_memory + 1`) to the node stats.

https://github.com/sirensolutions/siren-platform/blob/c520df9b14b5d928a0c5f07105e9102730b10ba8/core/src/main/java/io/siren/federate/core/io/memory/BufferAllocatorService.java#L132-L143

This way I could retrieve the value from an integ test. Do you think that's an interesting stat to add there ?

Reply:
The (small) issue I see is that this new stat will kind of duplicate the max direct memory stat, so its benefit is unclear to me.
At some point in the future, we might decide to remove the stat about max direct memory limit, and instead put the root memory limit (and maybe the others, i.e., job and task).

Reply:
where is the duplication ?

Reply:
From https://github.com/sirensolutions/siren-platform/blob/master/doc/src/main/asciidoc/modules/siren-federate/pages/cluster-apis.adoc#memory-information :

```json
    "memory" : {
      "allocated_direct_memory_in_bytes" : 0,
      "allocated_root_memory_in_bytes": 0,
      "root_allocator_dump": "Allocator(ROOT) 0/0/3750232064/17179869184 (res/actual/peak/limit)"
    }
```

Ah did you mean I could get the limit from `root_allocator_dump` ? I had forgotten about this, that is enough for my test!